### PR TITLE
Marquee: fix for wrong marquee position after reset

### PR DESCRIPTION
**Pull request by TomaszLukawskiSam**
[Issue] https://github.com/Samsung/TAU/issues/489
[Problem] Marquee: wrong marquee postion after reset & (re)start widget
[Solution] Animation reset has been implemented

Signed-off-by: Tomasz Lukawski <t.lukawski@samsung.com>

### CODE REVIEW

File: src/js/core/widget/core/Marquee.js [hunk 1]
```diff
@@ -642,12 +639,12 @@
...
+					animation = self._animation;
 
-				this.option("animation", "stopped");
-				stateDOM.style.webkitMaskImage = (this.options.ellipsisEffect === "none") ? "" : GRADIENTS.RIGHT;
-				stateDOM.children[0].style.webkitTransform = "translateX(0)";
-				self._render(true);
+				animation.reset();
+
+				self.element.style.webkitMaskImage = (self.options.ellipsisEffect === "none") ?
+					"" : "-webkit-linear-gradient(left, #ffffff 0, #ffffff 85%, transparent 100%)";
```
Comment (pkaczmarczy) [inline src/js/core/widget/core/Marquee.js]: Could you please use gradient style settings from GRADIENTS object?

File: src/js/core/util/animation/animation.js [hunk 2]
```diff
@@ -283,11 +283,49 @@
...
+			}
+
+			/**
+			 * Reset animations to initial position
+			 */
+			prototype.reset = function () {
+				var self = this;
+
+				if (self.active) {
+					// reset durring animation
```
Comment (pkaczmarczy) [inline src/js/core/util/animation/animation.js]: durring -> during

File: src/js/core/util/animation/animation.js [hunk 3]
```diff
@@ -283,11 +283,49 @@
 				}
 			};
 
+			/**
+			 * Method resets startTime for each animations to current time
+			 * @private
+			 * @param {*} animateConfig
+			 */
+			function resetStartTimeForAnimateConfig(animateConfig) {
```
Comment (pkaczmarczy) [inline src/js/core/util/animation/animation.js:295]: I am getting the exception during scrolling on wearable profile:
Uncaught TypeError: Cannot convert undefined or null to object
    at Function.keys (<anonymous>)
    at resetStartTimeForAnimateConfig (tau.js:18923)
    at Animate.prototype.reset (tau.js:18943)
    at Marquee.prototype.reset (tau.js:19711)
    at Listview.prototype._onChange (tau.js:32417)
    at Listview.prototype.handleEvent (tau.js:32683)
    at Object.trigger (tau.js:1273)
    at Listview.prototype.trigger (tau.js:8119)
    at Listview.prototype._rollDown (tau.js:32075)
    at Listview.prototype._onRotary (tau.js:32133)

**Review by pkaczmarczy — CHANGES_REQUESTED** [review]

**Review by pkaczmarczy — CHANGES_REQUESTED** [review]
After callining

widget.reset(); widget.stop(); synchronously one after another Marquee does not reset.
If called in widget.stop() then widget.reset() everything is fine

File: src/js/core/util.js [hunk 4]
```diff
@@ -131,6 +131,39 @@
 				window.clearTimeout(currentFrame);
 			};
 
+			/**
+			 * Remove animation callbacks added by requestAnimationFrame
+			 * @method cancelAnimationFrames
+			 * @static
+			 * @member ns.util
+			 * @param {*} animationId value for identify animation in quee
```
Comment (lslachciak) [inline src/js/core/util.js]: typo "quee" -> queue
unfortunatelly it stopped CI :-(

**Comment by lslachciak** [comment]
@singa2000,  looks like we still need to make some improvements, sorry can't make release today

File: src/js/core/util.js (hunk 4 above)
Comment (TomaszLukawskiSam) [reply src/js/core/util.js]: Fixed.

**Comment by TomaszLukawskiSam** [comment]
> After callining
> 
> widget.reset(); widget.stop(); synchronously one after another Marquee does not reset.
> If called in widget.stop() then widget.reset() everything is fine

Fixed.

**Review by pkaczmarczy — APPROVED** [review]
Looks good to me.

**Pull request merged**